pkg/util/runtime/runtime_linux.go

```diff
@@ -29,7 +30,14 @@ func parseKernelVersionStr(kernelVersionStr string) (semver.Version, error) {
 	// Patch: 0
 	// Pre: [72-generic]
 	// Build: []
-	return semver.Parse(kernelVersionStr)
+	verStrs := strings.Split(kernelVersionStr, ".")
+	switch {
+	case len(verStrs) < 2:
+		return semver.Version{}, fmt.Errorf("unable to get kernel version from %q", kernelVersionStr)
+	case len(verStrs) < 3:
+		verStrs = append(verStrs, "0")
```

Comment:
Does this mean "5.0" is also legal but "5" is not legal? I thought it always has at least major, minor and patch number but I'm not sure.
If you are not sure either, maybe we can just check if the length is less than 3 and return error if yes.

Reply:
Some special versions may not have patch number,  such as: "5.14-rc2" in https://www.kernel.org/, so need special handling :)

Reply:
thanks for the information